src/dmd/mtype.d

```diff
@@ -470,6 +469,18 @@ extern (C++) abstract class Type : ASTNode
 
     extern (C++) __gshared Type[TMAX] basic;
 
+    /// Returns the resolved special va_list type.
+    extern (C++) static Type getVaList(Scope* sc)
+    {
+        // Target.va_listType() may return a type based on some TypeIdentifier,
+        // possibly referring to a symbol in core.stdc.stdarg etc.
+        // So defer its semantic by resolving va_list lazily.
+        __gshared Type tvalist = null;
```

Comment:
Not sure what the current status wrt. avoiding global state for DMD-as-a-library etc. is. I'm all ears if there's a better place for this to live.

Reply:
There are a few places where this cached pointer is used, and is also initialized directly from the glue during language init. But there is no strong dependency on it remaining though (target.va_listType can always be used instead).